src/cryptography/x509.py

```diff
@@ -1344,3 +1344,52 @@ def extensions(self):
         """
         Returns an Extensions object containing a list of Revoked extensions.
         """
+
+
+class CertificateSigningRequestBuilder(object):
+    def __init__(self, version=Version.v1, subject_name=None, extensions=[]):
+        """
+        Creates an empty X.509 certificate request (v1).
+        """
+        self._version = Version.v1
+        self._subject_name = subject_name
+        self._extensions = extensions[:]
+
+    def set_version(self, version):
```

Comment:
Actually, the ones I added recently are all generated with version 3 (0x2).  If the code that loads CSRs is supposed to raise when parsing "v3" certificates, the code is probably broken because those CSRs load fine in the test suite.

I'll let you check whether that code should be fixed.  Should I go an fix the test vectors that are incorrectly set to v3 (and then remove the `set_version()` method from the builder)?